Organize Project: Populate Job Data and other docs

johnbradley opened the pull request:
Improvements by copying cwlreport/scriptsreadme classes from [lando worker ](https://github.com/Duke-GCB/lando/tree/ae65f9a21bda5be77a1aa34e481da163df2a1f8d/lando/worker). Also includes tests from that repository.

Moves lando_util/organize_project.py to lando_util/organize_project/organizer.py. NOTE: Github isn't showing this rename but git is [aware of the rename](https://github.com/Duke-GCB/lando-util/pull/6/commits/9eda745f4c9ec24d582e1ee4ea46d6135e041b8c).

### Changes to `organize_project` input command file
Removes `job_data_path` property. This data can be mostly derived by the organize project command.
Adds `bespin_job_id`, `bespin_workflow_started` and `bespin_workflow_finished` properties.
The `bespin_workflow_started` and `bespin_workflow_finished` properties are optional for now since these values are not stored in bespin-api currently and may be difficult to determine when restarting organize output.

Fixes #2

Created issue in lando for lando_worker to use `lando-util`: https://github.com/Duke-GCB/lando/issues/154

johnbradley committed:
move organize_project into a module

johnbradley committed:
move organize_project tests to new module

johnbradley committed:
copy document generation classes from lando worker

https://github.com/Duke-GCB/lando/blob/075b08e9ae1801fe25ddeeb49ef43a0c968a5bcf/lando/worker/scriptsreadme.py
https://github.com/Duke-GCB/lando/blob/075b08e9ae1801fe25ddeeb49ef43a0c968a5bcf/lando/worker/cwlreport.py

johnbradley committed:
rename organizer.py

johnbradley committed:
changes to use classes from lando

johnbradley committed:
add main to keep organize project command consistent

johnbradley committed:
remove old organize_project tests

johnbradley committed:
add requirements for lando worker classes

johnbradley committed:
test that job_data is written as a JSON file

johnbradley committed:
test that ProjectData creates job_data property

johnbradley committed:
test write_data_to_file

dleehr reviewed (changes requested):
Thanks for clarifying the rename. I compared by diffing against 4cc6261dc79ab35624713b8c6018a29d6742bd49

Based on the feedback and [issues](https://github.com/Duke-GCB/lando/milestone/2) we have for the  Jinja/Markdown lando-generated reports/documentation, I think we're due to address those. While there can be a lot of inertia in existing code (in the lando codebase), I think this lando-util codebase is an opportunity to start on that effort, and would prefer not to copy these forward.

I'd suggest we strip this down to the bare minimum, and build back up:

- artifacts for reproducibility (workflow, job order doc, log, output object)
- computation metrics (runtime, disk space, etc)
- whatever bespin-api requires (a simple README?)

Apologies for not catching this as the task of #2 sooner. I thought we had a bigger issue for cleaning up the reports, but I'll create one.

dleehr commented:
The changes for https://github.com/Duke-GCB/lando/pull/155 are good though.

johnbradley committed:
simplify organize output

Removes methods document and script README.
Renames CwlReport to ReadmeReport.
Simplifies top level README.

johnbradley committed:
finish removing methods template

johnbradley commented:
I removed the scripts directory README and the Methods document. Also simplified the top level docs/README.
